### PR TITLE
Remove isSampled, follow up from #1449

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/Sampler.java [hunk 1]
```diff
@@ -91,13 +91,11 @@ SamplingResult shouldSample(
...
 
-    boolean isSampled();
-
     /**
      * Return tags which will be attached to the span.
      *
-     * @return attributes added to span. These attributes should be added to the span only for root
-     *     span or when sampling result {@link #isSampled()} changes from false to true.
+     * @return attributes added to span. These attributes should be added to the span only when
+     *     sampling decision {@link #getDecision()} is {@code RECORD} or {@code RECORD_AND_SAMPLED}.
```
Comment (anuraaga) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/Sampler.java:98]:
```suggestion
     *     {@linkplain #getDecision() the sampling decision} is {@link SamplingDecision#RECORD} or {@link SamplingDecision#RECORD_AND_SAMPLED}.
```

File: sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java [hunk 2]
```diff
@@ -65,13 +65,17 @@
...
+  }
+
+  static boolean isSampled(Decision decision) {
+    return Decision.RECORD_AND_SAMPLED.equals(decision);
   }
 
   /**
    * Returns a {@link SamplingResult} with the given {@code attributes} and {@link
-   * SamplingResult#isSampled()} returning {@code isSampled}.
+   * SamplingResult#getDecision()} ()} returning {@code decision}.
```
Comment (anuraaga) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java:78]:
```suggestion
   * SamplingResult#getDecision()} returning {@code decision}.
```

Comment (anuraaga) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java:78]: Also tend to find it more readable to use something like `the provided {@link Decision}`.

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/Samplers.java]: Done.

**Pull request merged**